nixos/modules/services/misc/gitlab.nix

```diff
@@ -381,6 +389,34 @@ in {
         '';
       };
 
+      registry = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Enable gitLab container registry.";
+        };
+        host = mkOption {
+          type = types.str;
+          default = config.services.gitlab.host;
+          description = "GitLab container registry host name.";
+        };
+        port = mkOption {
+          type = types.int;
+          default = 4567;
+          description = "GitLab container registry port.";
+        };
+        certFile = mkOption {
+          type = types.path;
+          default = null;
+          description = "Path to gitLab container registry certificate.";
+        };
+        keyFile = mkOption {
+          type = types.path;
+          default = null;
+          description = "Path to gitLab container registry certificate-key.";
+        };
```

Comment:
see my note from above regarding duplication of configuration options.

Options not existing there but set in the gitlab module should probably be added to the dockerRegistry module.

Reply:
These are the options, which control the endpoint, which is presented to the user. I think this should stay there.